include/zephyr/drivers/regulator/consumer.h

```diff
@@ -106,16 +106,61 @@ int regulator_get_current_limit(const struct device *dev);
  * @brief Select mode of regulator
  * Regulators can support multiple modes in order to permit different voltage
  * configuration or better power savings. This API will apply a mode for
- * the regulator, and also configure the remainder of the regulator APIs,
- * such as those disabling, changing voltage/current targets, or querying
- * voltage/current targets to target that mode.
+ * the regulator.
  * @param dev: regulator to switch mode for
  * @param mode: Mode to select for this regulator. Only modes present
  * in the regulator-allowed-modes property are permitted.
  * @return 0 on success, or errno on error
  */
 int regulator_set_mode(const struct device *dev, uint32_t mode);
 
+/**
+ * @brief Set target voltage for regulator mode
+ * Part of the extended regulator consumer API.
+ * sets the target voltage for a given regulator mode. This mode does
+ * not need to be the active mode. This API can be used to configure
+ * voltages for a mode, then the regulator can be switched to that mode
+ * with the regulator_set_mode api.
+ * @param dev: regulator to set voltage for
+ * @param mode: target mode to configure voltage for
+ * @param min_uV: minimum voltage acceptable, in uV
+ * @param max_uV: maximum voltage acceptable, in uV
+ * @return 0 on success, or errno on error
+ */
+int regulator_set_mode_voltage(const struct device *dev, uint32_t mode,
+	uint32_t min_uV, uint32_t max_uV);
+
+/**
+ * @brief Get target voltage for regulator mode
+ * Part of the extended regulator consumer API.
+ * gets the target voltage for a given regulator mode. This mode does
+ * not need to be the active mode. This API can be used to read voltages
+ * from a regulator mode other than the default.
+ * @param dev: regulator to query voltage from
+ * @param mode: target mode to query voltage from
+ * @return voltage level in uV
+ */
+int regulator_get_mode_voltage(const struct device *dev, uint32_t mode);
+
```

Comment:
Can we get rid of the `regulator_set_voltage` and `regulator_get_voltage` API's since they seem similar.

Reply:
We do still need these APIs- since mode setting support is optional, regulators without modes will use these APIs over the new mode-aware ones